src/api/v1/controller.ts

```diff
@@ -82,11 +85,23 @@ export class Controller {
         }
 
         // broadcast transaction
-        const result = await broadcastTransaction(signedTx, network);
+        let result = await broadcastTransaction(signedTx, network);
+
+        if (result?.reason === TxRejectedReason.BadNonce) {
+          req.logger.warn({ tx, nonce, result }, 'Bad nonce. Will update cached nonce and try again');
+          const address = getAddressFromPrivateKey(account.stxPrivateKey, TransactionVersion.Mainnet);
+          await setupAccountNonce(address);
+          result = await broadcastTransaction(signedTx, network);
+        }
 
         // increment nonce or handle errors
         if ('error' in result) {
-          throw new Error(`Broadcast failed: ${result.error}`);
+          req.logger.warn({
+            tx,
+            nonce,
+            result,
+          });
```

Comment:
Looks good. This is the only thing I'd change maybe; just add a message for context as to what failed.

```suggestion
          req.logger.warn({
            tx,
            nonce,
            result,
          }, 'failed to broadcast transaction');
```